src/sql/base/browser/ui/table/table.ts

```diff
@@ -121,6 +123,58 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 		this.mapMouseEvent(this._grid.onHeaderClick, this._onHeaderClick);
 		this.mapMouseEvent(this._grid.onDblClick, this._onDoubleClick);
 		this._grid.onColumnsResized.subscribe(() => this._onColumnResize.fire());
+
+		this._grid.onClick.subscribe((e, data) => {
+			if (this.isTreeGrid()) {
+				this.toggleTreeGridParent(data.row, data.cell);
+			}
+		});
+
+		this._grid.onKeyDown.subscribe((e, data) => {
+			if (this.isTreeGrid()) {
+				if ((<any>e).keyCode === 13) {
+					// When we press enter on an collapsed cell we expand it.
+					this.toggleTreeGridParent(data.row, data.cell);
+				} else if ((<any>e).keyCode === 37) {
+					// Left arrow on first cell of the expanded row collapses it
+					if (data.cell === 0) {
+						this.toggleTreeGridParent(data.row, this.expandableColumnIndex(), false);
+					}
+				} else if ((<any>e).keyCode === 39) {
+					// Right arrow on last cell of the collapsed row expands it.
+					if (data.cell === (this._grid.getColumns().length - 1)) {
+						this.toggleTreeGridParent(data.row, this.expandableColumnIndex(), true);
+					}
+				}
+			}
+		});
+
+		this._grid.onRendered.subscribe((e, data) => {
+			if (this.isTreeGrid()) {
+				// Changing table role from grid to treegrid
+				this._tableContainer.setAttribute('role', 'treegrid');
+				for (let i = 0; i < this._data.getLength(); i++) {
+					const rowData = this._data.getItem(i);
+					// Getting the row div that corresponds to the data row
+					const rowElement = this._tableContainer.querySelector(`div [role="row"][aria-rowindex="${(i + 1)}"]`);
+					// If the row element is found in the dom, we are setting the required aria attributes for it.
+					if (rowElement) {
```

Comment:
is the row html element not being reused by slickgrid? if it is you also need to clear the values you set previously for a different row.

Reply:
It does not really matter because I am changing the attributes for all the visible rows. So, even if they are reused, my for loop should take care of it.

Reply:
but you do have null/undefined check, so it is possible that you are not clearing the attribute that were set previously.

Reply:
nvm, it looks like you are setting the values for every row in transform. so that should be good.